ui/src/app/store/pod/selectors.ts

```diff
@@ -227,6 +240,34 @@ const getVmResource = createSelector(
   }
 );
 
+/**
+ * Returns a pod's storage pools, sorted by default first then id.
+ * @param state - The redux state.
+ * @param podId - The id of the pod.
+ * @returns A list of the pod's storage pools, sorted by default first then id.
+ */
+const getSortedPools = createSelector(
+  [
+    (state: RootState, podId: Pod["id"]) =>
+      defaultSelectors.getById(state, podId),
+  ],
+  (pod) => {
+    if (!pod) {
+      return [];
+    }
+    const pools = pod.storage_pools || [];
+    return [...pools].sort((a, b) => {
```

Comment:
I don't think this spread is necessary here, Immer should guarantee that the store is not updated. Was this a precaution or was something odd going on?

Reply:
This was just a copy+paste from within the original component so it might not be necessary any longer. Let me remove it and see if anything changes

Reply:
Seems to be fine!